File: perception/sensors/camera.py
import cv2 as cv

class Camera:
    def __init__(self, url):
        self.url = url
    def save_image(self, filename):
        capture = cv.VideoCapture(self.url)
        if not capture.isOpened():
            print("Cannot open camera")
            exit()
        ret, frame = capture.read()
        if not ret:
            print("Can't receive frame (stream end?). Exiting ...")
        cv.imwrite(filename, frame)
        capture.release()

File: docker/camera/camera.cpp
#include "camera.h"

VideoCapture camera(char* url) {
    return new cv::VideoCapture();
}

void save_image(VideoCapture camera, const char* filename)
{
    cv::Mat* image = new cv::Mat();
    ((cv::VideoCapture*)camera)->read(*image);
    cv::imwrite(filename, *image);
}
